src/test/java/DesignAssembly/DesignTest.java

```diff
@@ -0,0 +1,311 @@
+     * @param cell the (LUT3) cell to test
+     */
+    protected void testLUT3(Cell cell) {
+        // assert that the cell really is a LUT3
+        assertEquals("LUT3", cell.getLibCell().getName(), "LUT3 cell has wrong library cell attached");
+        // a LUT3 cell should have 3 inputs, and 1 output
+        assertEquals(3, cell.getInputPins().size(), "LUT3 cell has incorrect input pin count");
+        assertEquals(1, cell.getOutputPins().size(), "LUT3 cell has incorrect output pin count");
+        // the INIT property of a LUT3 should have a length of 11 (8 for the bits, 3 for the number type
+        assertNotNull(cell.getProperties().get("INIT"), "Cannot find INIT property for LUT3 cell");
+        assertEquals(11, cell.getProperties().get("INIT").getStringValue().length(), "LUT3 cell has incrrect INIT property");
+        for (CellPin pin : cell.getInputPins()) {
+            // each input pin should begin with "I"
+            assertEquals("I", pin.getName().substring(0, 1), "LUT3 cell has incorrect input pins");
+        }
+        for (CellPin pin : cell.getOutputPins()) {
+            // the only output pin should be named "O"
+            assertEquals("O", pin.getName(), "LUT3 cell has incorrect output pin");
+        }
+    }
+
+    /**
+     * test an IPORT cell's input and output pins
+     * @param cell the (IPORT) cell to test
+     */
+    protected void testIPORT(Cell cell) {
+        // assert that the cell really is an IPORT
+        assertEquals("IPORT", cell.getLibCell().getName(), "IPORT cell has wrong library cell attached");
+        // an IPORT cell should have no inputs and one output
+        assertEquals(0, cell.getInputPins().size(), "IPORT cell has incorrect input pin count");
+        assertEquals(1, cell.getOutputPins().size(), "IPORT cell has incorrect output pin count");
+        // test that the output pin is a "PAD"
+        for (CellPin pin : cell.getOutputPins()) {
+            assertEquals("PAD", pin.getName(), "IPORT cell has incorrect output pin");
+        }
+    }
+
+    /**
+     * test an OPORT cell's input and output pins
+     * @param cell the (OPORT) cell to test
+     */
+    protected void testOPORT(Cell cell) {
+        // assert that the cell really is an OPORT
+        assertEquals("OPORT", cell.getLibCell().getName(), "OPORT cell has wrong library cell attached");
+        // an OPORT cell should have one input and no outputs
+        assertEquals(1, cell.getInputPins().size(), "OPORT cell has incorrect input pin count");
+        assertEquals(0, cell.getOutputPins().size(), "OPORT cell has incorrect output pin count");
+        // test that the input pin is a "PAD"
+        for (CellPin pin : cell.getInputPins()) {
+            assertEquals("PAD", pin.getName(), "OPORT cell has incorrect input pin");
+        }
+    }
+
+    /**
+     * test a Flip Flop cell's properties
+     * @param cell the (Flip Flop) cell to test
+     */
+    protected void testFF(Cell cell) {
```

Comment:
I guess I should clarify my last comment. Is there any reason why these aren't private?